src/release/breaking-changes/android-14-nonlinear-text-scaling.md

```diff
@@ -0,0 +1,328 @@
+| `EditableText({ double? TextScaleFactor = 1.0 })` constructor argument  | 'textScaleFactor' is deprecated and shouldn't be used.        |
+| `EditableText.textScaleFactor` getter                                   | 'textScaleFactor' is deprecated and shouldn't be used.        |
+
+### Material library
+
+| Affected APIs                                                                 | Error Message                                          |
+|-------------------------------------------------------------------------------|--------------------------------------------------------|
+| `SelectableText({ double? TextScaleFactor = 1.0 })` constructor argument      | 'textScaleFactor' is deprecated and shouldn't be used. |
+| `SelectableText.rich({ double? TextScaleFactor = 1.0 })` constructor argument | 'textScaleFactor' is deprecated and shouldn't be used. |
+| `SelectableText.textScaleFactor` getter                                       | 'textScaleFactor' is deprecated and shouldn't be used. |
+
+## Migration guide
+
+Widgets provided by the Flutter framework are already migrated. Migration is 
+needed only if you're using any of the deprecated symbols listed above. 
+
+However, the new Android 14 nonlinear scaling feature changes the UI of existing 
+apps, even if your app does not directly use any of the deprecated APIs. If your 
+app targets Android 14 devices, it is **strongly recommended** to test your app 
+on Android 14 devices at different scaling levels, to make sure the UI is usable 
+and functional, no matter whether the codebase needs migration or not. Please 
+refer to the [Ensure Android 14 compatibility][] guide for steps to make sure 
+your app is Android 14 compatible.
+
+As an example, the following widget might not look great on Android 14 at 200% 
+scaling:
+
+```dart 
+// It's not good for tooltips to be cutoff vertically, or to take too much space 
+// on screen as it blocks contents.
+MyTooltipBox( 
+  size: tooltipBoxSize * textScaleFactor,
+  child: RichText(..., style: TextStyle(fontSize: 20)),
+)
+```
+
+With nonlinear scaling at 200%, the actual font size used by the `RichText` 
+widget is 21 (instead of 40), and the `MyTooltipBox` widget would take 
+significantly more screen space than needed, potentially making the tooltip 
+annoying. On the other hand, at a smaller font size, the tooltip's text could be 
+cutoff due to insufficient vertical space.
+
+Adopting nonlinear text scaling might not be an easy task. If you're unsure about 
+how to migrate certian `textScaleFactor` usages, or when it is going to take an
+extended period of time (for example, when the UI has to be redesigned), 
+consider replacing `textScaleFactor` with `TextScaler.textScaleFactor` first. The 
+`TextScaler.textScaleFactor` getter was added for backward compatibility during 
+the migration process and is expected to have a longer lifetime than other 
+`textScaleFactor` deprecations. 
```

Comment:
I don't think we should promise a longer deprecation period. The regular policy applies here (for now).